Realm/RLMProperty.mm

```diff
@@ -343,8 +381,11 @@ - (instancetype)initWithName:(NSString *)name
 
     _name = name;
     _indexed = indexed;
+    _objectClassName = linkingObjectsInfo[@"class"];
```

Comment:
These should be validated to be present within the dictionary and of the expected type.